### PR TITLE
refactor/extend tests for islRttSpec

### PR DESCRIPTION
**Pull request by andriidovhan**

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy [hunk 1]
```diff
@@ -1,358 +1,649 @@
...
+                assert !islUtils.getIslInfo(newIsl).isPresent()
+                assert !islUtils.getIslInfo(newIsl.reversed).isPresent()
             }
         }
+        database.resetCosts()
+    }
+
+    @Tidy
+    @Tags([HARDWARE])
+    def "ISL RTT stats are available for one direction in case link is UP in forward direction only"() {
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy:277]: Can you review this test? It seems that test name does not correlate with the actual behavior

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy [hunk 2]
```diff
@@ -1,358 +1,649 @@
...
+        changeIslRttToggle(true)
+
+        then: "IslRtt rules are installed"
+        wait(RULES_INSTALLATION_TIME) { checkIslRttRules(sw.dpId, true) }
+
+        when: "Update server42Port on the switch(revert to origin)"
+        northbound.updateSwitchProperties(sw.dpId, northbound.getSwitchProperties(sw.dpId).jacksonCopy()
+                .tap({ it.server42Port = originSwProps.server42Port }))
+
+        then: "SERVER_42_ISL_RTT_OUTPUT_COOKIE and SERVER_42_ISL_RTT_INPUT rules updated according to the changes"
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy:439]: How do you know which exactly state is 'origin' to do the check?

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy [hunk 3]
```diff
@@ -1,358 +1,649 @@
...
         checkpointTime = new Date()
-        TimeUnit.SECONDS.sleep(statsWaitSeconds)
+        def swIsConnected = true
 
-        then: "Expect no ISL RTT forward stats"
-        otsdb.query(checkpointTime, metricPrefix + "isl.rtt",
-                [src_switch: isl.srcSwitch.dpId.toOtsdFormat(),
-                 src_port  : String.valueOf(isl.srcPort),
-                 origin    : "server42"]).dps.isEmpty()
+        then: "ISL Rtt rules are still exist on the src switch"
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy:554]: `rules still exist` or `rules are still present`

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy:439]: I retrieve it before.
I mean that I found the switch which is connected to s42 and got swProps from it `def originSwProps = northbound.getSwitchProperties(sw.dpId)`. Assume that this data is correct(origin).

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy (hunk 3 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy]: `rules still exist`

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy [hunk 4]
```diff
@@ -1,358 +1,649 @@
...
-            }
+        then: "ISL RTT forward stats are available"
+        and: "ISL RTT reverse stats are available"
+        wait(islSyncWaitSeconds, 2) {
+            checkIslRttStats(isl, checkpointTime, true)
+            checkIslRttStats(isl.reversed, checkpointTime, true)
+        }
+
+        and: "ISL latency value is updated in db in both direction from s42"
+        wait(300, 2) { //latency.update.interval=300 TODO(andriidovhan) rework, probably reduce to 3m
```
Comment (sergii-iakovenko) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy:161]: Do you mean 3 seconds not 3m?

**Review by sergii-iakovenko — APPROVED** [review]

**Review by rtretyak — APPROVED** [review]

**Pull request merged**